### PR TITLE
test(client): auth session tests

### PR DESCRIPTION
**Pull request by gao-sun**
## Summary

`.xcscheme` files are auto-generated by Xcode when enabling test coverage.

- [x] enable test coverage for all targets
- [x] add `LogtoWebAuthSession` protocol for testing injection
- [x] init `LogtoAuthSession` test framework

## Testing

the change itself

## Reviewers

@logto-io/eng

### CODE REVIEW

File: Tests/LogtoClientTests/LogtoAuthSession/LogtoAuthSessionTests.swift [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        self.completionHandler = completionHandler
+    }
+
+    func start() -> Bool {
+        guard url.path != "/canceled" else {
+            completionHandler(nil, ASWebAuthenticationSessionError(.canceledLogin))
+            return true
+        }
+
+        completionHandler(redirectUri, nil)
```
Comment (simeng-li) [inline Tests/LogtoClientTests/LogtoAuthSession/LogtoAuthSessionTests.swift:23]: nit: would it be better to use a positive statement if possible?:
```
guard url.path = "/canceled" else {
  completionHandler(redirectUri, nil);
  return truel
}

completionHandler(nil, ASWebAuthenticationSessionError(.canceledLogin))
 return true
```

**Review by simeng-li — APPROVED** [review]
overall LGTM

File: Tests/LogtoClientTests/LogtoAuthSession/LogtoAuthSessionTests.swift (hunk 1 above)
Comment (gao-sun) [reply Tests/LogtoClientTests/LogtoAuthSession/LogtoAuthSessionTests.swift:24]: yes, but we got more conditions to go. i'll update to switch clause

Comment (gao-sun) [reply Tests/LogtoClientTests/LogtoAuthSession/LogtoAuthSessionTests.swift]: updated, PTAL

**Pull request merged**